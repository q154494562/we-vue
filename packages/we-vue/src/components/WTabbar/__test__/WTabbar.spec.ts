import { shallowMount } from '@vue/test-utils'
import Tabbar from '../'

describe('tabbar', () => {
  test('create', () => {
    const wrapper = shallowMount(Tabbar, {
      propsData: {},
    })

    expect(wrapper.name()).toBe('w-tabbar')
    expect(wrapper.classes()).toContain('weui-tabbar')

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('fixed tabbar', () => {
    // is not fixed by default
    let wrapper = shallowMount(Tabbar, {
      propsData: {},
    })

    expect(wrapper.element.style.position).toBe('absolute')

    wrapper = shallowMount(Tabbar, {
      propsData: {
        fixed: true,
      },
    })

    expect(wrapper.element.style.position).toBe('fixed')
  })
})
