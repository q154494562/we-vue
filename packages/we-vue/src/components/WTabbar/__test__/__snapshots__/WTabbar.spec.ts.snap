// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tabbar create 1`] = `<div class="weui-tabbar" style="position: absolute;"></div>`;
